### PR TITLE
Solutions

### PR DESCRIPTION
**Pull request by patjdor**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,4 +6,4 @@ This is a small repository of coding puzzles. To initialize:
 $ git@github.com:iabaldwin/quizzy.git
 ```
 
-See the Makefile for a descriptrion of each of the problems.
+See the Makefile for a description of each of the problems.
```
Comment (ryancdougherty) [inline README.md:9]: He caught some typos and corrected them.

File: entropy.cc [hunk 2]
```diff
@@ -1,20 +1,99 @@
...
+    
+    // Configure histogram
+    // An example configuration:
+    //      Discretization: 8
+    //      Input Range: [0,128)
+    //      Bins: { 0 = [0,8), 1 = [8,16), ... 15 = [120,128) }
+    int num_bins = 128/Discretization;
+    int interval = Discretization;
+    unsigned int histogram[num_bins];
+    std::memset(histogram,0,sizeof(histogram));
```
Comment (ryancdougherty) [inline entropy.cc:47]: Is this normal? Wouldn't `unsigned int histrogram[num_bins] = {0};` also work?

File: entropy.cc [hunk 3]
```diff
@@ -1,20 +1,99 @@
...
+    
+    int sample_count = 0;
+    
+    for (unsigned int ii = 0; ii < NUM_SAMPLES; ++ii){
```
Comment (ryancdougherty) [inline entropy.cc:51]: Why do people use this double i: `ii`? Psiaki always did that to loop indices.

File: strings.cc [hunk 4]
```diff
@@ -1,23 +1,39 @@
...
+ * "N" a word?). All characters in the string are treated as part of the word; 
+ * e.g. "a man a plan a canal panama" is a palindromic sentence, but only
+ * after adjusting the spacing.
+ * 
+ * The solution is O(N) in the number of characters. It uses std::equal to 
+ * compare the first half of the word to the second half in reverse order.
+ * @see std::equal
+ */
+inline bool isPalindromic(const std::string& word){
+    return std::equal(word.begin(), word.begin()+word.size()/2, word.rbegin());
```
Comment (ryancdougherty) [inline strings.cc:22]: Interesting: if `word.size()` is odd, there is no need to check the middle letter, and this code doesn't check it.

File: entropy.cc (hunk 3 above)
Comment (patjdor) [reply entropy.cc:51]: I learned this from a colleague a few years ago: try searching for "i" in a file; now try searching for "ii". It just makes it more convenient to locate.

**Pull request closed without merging**